persistence/repository.go

```diff
@@ -21,6 +21,7 @@ func NewDynamoRepository(config *config.Config, db *dynamodb.DynamoDB) *DynamoRe
 type Repository interface {
 	GetByID(string) (*Resource, error)
 	SaveItem(*Resource) error
```

Comment:
Since you've added `UpdateItem` it seems like it would be a good idea to change `SaveItem` to `CreateItem`.